src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleService.kt

```diff
@@ -66,28 +66,49 @@ class ScheduleService(
     requestedSchedules.filter { it.isFutureSchedule() }
       .map { requestedSchedule ->
         courseActivity.courseSchedules.filter { it.isFutureSchedule() }
-          .find { savedSchedule -> requestedSchedule matches savedSchedule }
-          ?.let { savedSchedule -> savedSchedule.apply { scheduleStatus = requestedSchedule.scheduleStatus } }
+          .find { savedSchedule -> requestedSchedule.courseScheduleId == savedSchedule.courseScheduleId }
+          ?.update(requestedSchedule)
           ?: let { requestedSchedule }
       }
       .toList()
 
   private fun CourseSchedule.isFutureSchedule() = scheduleDate > LocalDate.now()
 
-  // TODO SDIT-838 match by id once it is passed on the course schedule request
-  private infix fun CourseSchedule.matches(other: CourseSchedule) =
-    scheduleDate == other.scheduleDate &&
-      startTime == other.startTime &&
-      endTime == other.endTime
+  private fun CourseSchedule.update(requested: CourseSchedule): CourseSchedule {
+    checkForIllegalUpdate(requested)
+    return this.apply {
+      scheduleDate = requested.scheduleDate
+      startTime = requested.startTime
+      endTime = requested.endTime
+      slotCategory = requested.slotCategory
+      scheduleStatus = requested.scheduleStatus
+    }
+  }
+
+  private fun CourseSchedule.checkForIllegalUpdate(requested: CourseSchedule) {
+    if (scheduleDate <= LocalDate.now() &&
+      (
+        scheduleDate != requested.scheduleDate ||
+          startTime != requested.startTime ||
+          endTime != requested.endTime
+        )
+    ) {
+      throw BadDataException("Cannot update schedules starting before tomorrow")
+    }
+  }
```

Comment:
Retain this restriction for now to limit the size of this change - now we don't rely on the business key we can change it whenever, coming soon.